src/main/scala/com.snowplowanalytics.iglu/core/ParseError.scala

```diff
@@ -15,24 +15,33 @@ package com.snowplowanalytics.iglu.core
 /** Common error type for parsing core Iglu entities */
 sealed trait ParseError extends Product with Serializable {
   def code: String
+  def message(data: String): String
 }
 
 object ParseError {
   case object InvalidSchemaVer extends ParseError {
     def code = "INVALID_SCHEMAVER"
+    def message(data: String) = s"Invalid schema version $data, code: $code"
   }
   case object InvalidIgluUri extends ParseError {
     def code = "INVALID_IGLUURI"
+    def message(data: String) = s"Invalid Iglu URI $data, code: $code"
   }
   case object InvalidData extends ParseError {
     def code = "INVALID_DATA_PAYLOAD"
+    def message(data: String) = s"Invalid data payload $data, code: $code"
   }
   case object InvalidSchema extends ParseError {
     def code = "INVALID_SCHEMA"
+    def message(data: String) = s"Invalid schema $data, code: $code"
+  }
+  case object UnknownMetaschema extends ParseError {
+    def code = "UNKNOWN_METASCHEMA"
+    def message(data: String) = s"cannot recognize meta schema URI $data, code: $code"
   }
 
   def parse(string: String): Option[ParseError] =
-    List(InvalidSchemaVer, InvalidIgluUri, InvalidData, InvalidSchema).find { _.code == string }
+    List(InvalidSchemaVer, InvalidIgluUri, InvalidData, InvalidSchema, UnknownMetaschema).find { e => string.contains(e.code) }
```

Comment:
Why `contains`?

Reply:
What other alternative do we have for finding a substring within a string?

Reply:
The question is why do we look for a _substring_? We need to find a string in a list of strings.